### PR TITLE
deep-link-for-local-notifications

### PR DESCRIPTION
**Pull request by scoder1747**
Deep Linking to Local Notifications (Mentions, Likes, Reposts, DMs, Zaps).

https://user-images.githubusercontent.com/120697811/230225286-f37a5e24-08ba-432d-94ce-535cb78d1d5a.mov

### CODE REVIEW

File: damus/ContentView.swift [hunk 1]
```diff
@@ -156,6 +158,13 @@ struct ContentView: View {
                     EmptyView()
                 }
             }
+            // Navigation when clicking local notification
+            if nav_local_notif_active {
+                let thread = ThreadModel(event: ev_local_notif!, damus_state: damus_state!)
```
Comment (jb55) [inline damus/ContentView.swift:163]: please don't use !

File: damus/ContentView.swift [hunk 2]
```diff
@@ -156,6 +158,13 @@ struct ContentView: View {
                     EmptyView()
                 }
             }
+            // Navigation when clicking local notification
+            if nav_local_notif_active {
```
Comment (jb55) [inline damus/ContentView.swift:162]: I don't think you need this if statement

Comment (jb55) [reply damus/ContentView.swift:162]: maybe it should just be 

```swift
if let ev_local_notif {
}
```

File: damus/ContentView.swift [hunk 3]
```diff
@@ -468,6 +477,16 @@ struct ContentView: View {
...
         }
+        .onReceive(handle_notify(.local_notification)) { notif in
+            if let obj = notif.object as? String {
+                if obj == "dm_local_notification" {
+                    selected_timeline = .dms
+                } else if let notif = damus_state?.events.lookup(obj) {
+                    ev_local_notif = notif
+                    nav_local_notif_active = true
+                }
+            }
```
Comment (jb55) [inline damus/ContentView.swift:488]: prefer this style:

```suggestion
            guard let obj = notif.object as? String else {
                return
            }

            if obj == "dm_local_notification" {
                selected_timeline = .dms
            } else if let notif = damus_state?.events.lookup(obj) {
                ev_local_notif = notif
                nav_local_notif_active = true
            }
```

Comment (scoder1747) [reply damus/ContentView.swift]: a good call out!

File: damus/Models/HomeModel.swift [hunk 4]
```diff
@@ -548,7 +548,7 @@ class HomeModel: ObservableObject {
                 self.new_events = notifs
                 if  damus_state.settings.dm_notification,
                     let displayName = damus_state.profiles.lookup(id: self.incoming_dms.last!.pubkey)?.display_name {
```
Comment (jb55) [inline damus/Models/HomeModel.swift:550]: just noticed a ! here, let's fix that while we're here

```suggestion
              if  damus_state.settings.dm_notification,
                     let last = self.incoming_dms.last,
                     let displayName = damus_state.profiles.lookup(id: last.pubkey)?.display_name {
```

Comment (jb55) [reply damus/Models/HomeModel.swift:550]: I also thing this logic is wrong since I keep getting the wrong display name, but we can fix that after

Comment (scoder1747) [reply damus/Models/HomeModel.swift]: addressed, yeah I will also look into it in depth, thx

**Comment by jb55** [comment]
trying this again!

**Comment by jb55** [comment]
I changed the code a bit! it's mostly the same, just refactored a bunch of stuff

**Pull request closed without merging**